exchange/bitswap/bitswap.go

```diff
@@ -375,6 +373,10 @@ func (bs *Bitswap) ReceiveMessage(ctx context.Context, p peer.ID, incoming bsmsg
 			k := b.Cid()
 			log.Event(ctx, "Bitswap.GetBlockRequest.End", k)
 
+			for _, ses := range bs.SessionsForBlock(k) {
+				ses.receiveBlockFrom(p, b)
```

Comment:
Out of curiosity, what happens if the user adds the block to the local blockstore via the commandline while we're trying to fetch it from the network?